msgraph-developer-proxy-abstractions/ILogger.cs

```diff
@@ -1,10 +1,34 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+using Titanium.Web.Proxy.EventArguments;
 
 namespace Microsoft.Graph.DeveloperProxy.Abstractions;
 
+public enum MessageType {
+    Normal,
+    InterceptedRequest,
+    PassedThrough,
+    Warning,
+    Tip,
+    Failed,
+    Chaos,
+    Mocked
+}
+
+public class LoggingContext {
+    public SessionEventArgs Session { get; }
+
+    public LoggingContext(SessionEventArgs session)
+    {
+        Session = session;
+    }
+}
+
 public interface ILogger {
-    public void Log(string message);
+    public void LogRequest(string[] message, MessageType messageType, LoggingContext? context = null);
```

Comment:
`LogRequest` is used to log decorated messages related to intercepting and processing requests